compiler/codegen/library.cpp

```diff
@@ -162,35 +160,31 @@ const char* getLibraryExtension() {
   return "";
 }
 
-void openLibraryHelperFile(fileinfo* fi, const char* name, const char* ext,
-                           bool useTmpDir) {
+void ensureLibDirExists() {
+  if (libDir[0] == '\0') {
+    strncpy(libDir, "lib", sizeof("lib"));
+    libDir[sizeof("lib")] = '\0';
```

Comment:
strncpy will add the terminating null - in fact it will initialize all of the memory with nulls.

Reply:
I was thinking of `strncpy` but just misreading what you were passing as the length field.

Since this is setting the default, don't we know it won't overflow libDir?

sizeof("lib") is 8 bytes because it's a pointer. Will work, but probably confused.

I think this code should be:
``` c
  if (libDir[0] == '\0') {
    const char* dir = "lib";
    INT_ASSERT(strlen(dir) < sizeof(libDir));
    strcpy(libDir, dir);
  }
```

Reply:
Apparently sizeof("lib") is OK; see https://bytes.com/topic/c/answers/212229-sizeof-string-literal ; but I still think it's more reasonable to write the above.